javascripts/discourse/api-initializers/discourse-mermaid-theme-component.js

```diff
@@ -58,9 +59,55 @@ async function applyMermaid(element, key = "composer") {
       mermaid.dataset.processed = true;
       mermaid.querySelector(".spinner")?.remove();
     }
+
+    if (key === "composer") {
+      debounce(() => updateMarkdownHeight(mermaid, index), 500);
```

Comment:
That debounce is useless. If you add a `console.log` in `updateMarkdownHeight` you will see it's called constantly while typing in composer.

The reason for this is that debounce can't work with an anonymous function, as it recreates a new context each time, I can fix it for you if you want, let me know.

Reply:
yeah ... half useless, at least it delays the call... deanonymising this function is a pita :( can you fix for me and merge, would love to see the fix.

Reply:
Will look today